tensorflow_serving/model_servers/http_server.cc

```diff
@@ -160,9 +160,20 @@ class RestApiRequestDispatcher {
     string output;
     VLOG(1) << "Processing HTTP request: " << req->http_method() << " "
             << req->uri_path() << " body: " << body.size() << " bytes.";
-    const auto status =
+    auto status =
         handler_->ProcessRequest(req->http_method(), req->uri_path(), body,
                                  &headers, &model_name, &method, &output);
+
+    if (req->http_method() == "OPTIONS") {
+      absl::string_view origin_header = req->GetRequestHeader("Origin");
+      if (RE2::PartialMatch(origin_header, "https?://")) {
+        status = Status::OK();
+      } else {
+        status = errors::FailedPrecondition(
+            "Origin header not found in request: ", req->http_method());
```

Comment:
I feel we should ignore this case, and let the existing REST model to handle the non-cors options requests.

@netfs   thoughts?

===

Longer term we should add cors handling in httpserver and have the rest model to configure the policy.

Reply:
I think adding OPTION request check here is kinda weird. But, I think this is the easiest way because I can get access to `net_http::ServerRequestInterface* req` and get the request header for other simple checks.